src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -0,0 +1,306 @@
+      if (this.persistedLog) {
+        this.persistedLog.add(query.query);
+      }
+
+      this.props.onSubmit({ query: fromUser(query.query), language: query.language });
+    }
+  };
+
+  private onChange = (query: Query, dateRange?: TimeRange) => {
+    if (this.props.onChange) {
+      this.props.onChange({ query: fromUser(query.query), language: query.language }, dateRange);
+    }
+  };
+
+  private onQueryStringChange = (value: string) => {
+    this.setState({
+      isSuggestionsVisible: true,
+      index: null,
+    });
+
+    this.onChange({ query: value, language: this.props.query.language });
+  };
+
+  private onInputChange = (value: string) => {
+    this.onQueryStringChange(value);
+  };
+
+  private onClickInput = (event: React.MouseEvent<HTMLTextAreaElement>) => {
+    if (event.target instanceof HTMLTextAreaElement) {
+      this.onQueryStringChange(event.target.value);
+    }
+  };
+
+  // TODO: MQL consider moving language select language of setting search source here
+  private onSelectLanguage = (language: string) => {
+    // Send telemetry info every time the user opts in or out of kuery
+    // As a result it is important this function only ever gets called in the
+    // UI component's change handler.
+    this.services.http.post('/api/opensearch-dashboards/dql_opt_in_stats', {
+      body: JSON.stringify({ opt_in: language === 'kuery' }),
+    });
+
+    const newQuery = {
+      query: this.props.getQueryStringInitialValue?.(language) ?? '',
+      language,
+    };
+
+    const fields = this.props.settings.getQueryEnhancements(newQuery.language)?.fields;
+    const newSettings: DataSettings = {
+      userQueryLanguage: newQuery.language,
+      userQueryString: newQuery.query,
+      ...(fields && { uiOverrides: { fields } }),
+    };
+    this.props.settings?.updateSettings(newSettings);
+
+    const dateRangeEnhancement = this.props.settings.getQueryEnhancements(language)?.searchBar
+      ?.dateRange;
+    const dateRange = dateRangeEnhancement
+      ? {
+          from: dateRangeEnhancement.initialFrom!,
+          to: dateRangeEnhancement.initialTo!,
+        }
+      : undefined;
+    this.onChange(newQuery, dateRange);
+    this.onSubmit(newQuery, dateRange);
+  };
+
+  private initPersistedLog = () => {
+    const { uiSettings, storage, appName } = this.services;
+    this.persistedLog = this.props.persistedLog
+      ? this.props.persistedLog
+      : getQueryLog(uiSettings, storage, appName, this.props.query.language);
+  };
+
+  public onMouseEnterSuggestion = (index: number) => {
+    this.setState({ index });
+  };
+
+  textareaId = htmlIdGenerator()();
+
+  public componentDidMount() {
+    const parsedQuery = fromUser(toUser(this.props.query.query));
+    if (!isEqual(this.props.query.query, parsedQuery)) {
+      this.onChange({ ...this.props.query, query: parsedQuery });
+    }
+
+    this.initPersistedLog();
+    // this.fetchIndexPatterns().then(this.updateSuggestions);
```

Comment:
dead code?

Reply:
Didn't want to delete this code since this where we will get suggestions once query assist is added. Right now the suggestions is not wired up into the query editor but will be used once assistant work is added.

thanks for catching. added to this issue for tracking:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6957